posthog/warehouse/api/saved_query.py

```diff
@@ -180,7 +181,17 @@ def ancestors(self, request: request.Request, *args, **kwargs) -> response.Respo
 
             ancestors = ancestors.union(model_path.path[start:-1])
 
-        return response.Response({"ancestors": ancestors})
+        # Optimistically convert ancestor strings to UUIDs
+        ancestor_ids = set()
+        for ancestor in ancestors:
+            try:
+                ancestor_uuid = uuid.UUID(ancestor)
+                ancestor_ids.add(str(ancestor_uuid))
+            except ValueError:
+                ancestor_ids.add(ancestor)
+                continue
```

Comment:
I would rather have this be a function that we can map, e.g.
```python
def try_convert_to_uuid(s: str) -> uuid.UUID | str:
    try:
        return str(uuid.UUID(s))
    except ValueError:
        return  s
```

And then:
```python
ancestors = ancestors.union(map(try_convert_to_uuid, model_path.path[start:-1]))
```

It keeps things dry as we can re-use this in descendants too.

Reply:
done